apps/devhub/tests/test_views.py

```diff
@@ -2997,6 +3004,12 @@ def test_unique_name(self):
         eq_(r.context['new_addon_form'].non_field_errors(),
             ['This name is already in use. Please choose another.'])
 
+    def test_unlisted_name_not_unique(self):
+        """We don't enforce name uniqueness for unlisted add-ons."""
+        addon = addon_factory(name='xpi name')
+        addon.update(is_listed=False)
+        self.post()
```

Comment:
On Tue, Aug 04, 2015 at 01:12:06AM -0700, Mathieu Agopian wrote:
>We can't do that if we want to be able to test the error message. Or maybe we could by checking the content of the exception, but that wouldn't be very clean.
>We already know the alternate case works with the current `test_unique_name`, don't we? Or maybe I didn't understand your suggestion?

I guess I'm just mostly worried that you're relying on a behavior of `self.post()` here that isn't actually tested elsewhere for this view. Maybe I'm just being paranoid, but I'd be a lot more comfortable if the test for non-errors relied on the same behavior as the test for errors.